platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/counters/BackpressureBlocker.java

```diff
@@ -90,23 +90,15 @@ public boolean block() throws InterruptedException {
      */
     @Override
     public boolean isReleasable() {
-        while (true) {
-            final long currentCount = count.get();
-
-            if (currentCount >= capacity) {
-                // We've reached capacity, so we need to block.
-                return false;
-            }
-
-            final boolean success = count.compareAndSet(currentCount, currentCount + 1);
-            if (success) {
-                // We've successfully incremented the count, so we're done.
-                return true;
-            }
-
-            // We were unable to increment the count because another thread concurrently modified it.
-            // Try again. We will keep trying until we are either successful or we observe there is
-            // insufficient capacity.
+        final long resultingCount = count.incrementAndGet();
```

Comment:
I think these changes will benefit from introducing simple unit tests. At the very least, something that asserts the old `isReleasable` and new are the same should suffice.

Reply:
Should be considered for `develop`.